src/services/ImportContactsService.ts

```diff
@@ -1,16 +1,13 @@
 import { Readable } from 'stream';
-import csvParse from 'csv-parse';
 
 import Tag from '@schemas/Tag';
 import Contact from '@schemas/Contact';
 
+import parseCSVAsync from '../utils/parseCSVAsync';
+
 class ImportContactsService {
   async run(contactsFileStream: Readable, tags: string[]): Promise<void> {
-    const parsers = csvParse({
-      delimiter: ';',
-    });
-
-    const parseCSV = contactsFileStream.pipe(parsers);
+    const emails = await parseCSVAsync(contactsFileStream);
```

Comment:
Eu moveria esta linha para antes do `await Promise.all` para o código ficar mais agrupado. O que acha?

Reply:
Acho que deveria rever bem a alteracao feita @VictorAssis, a linha ja se encontra antes.

Reply:
Eu digo imediatamente antes @overlineink , na linha 26, pois está lendo o arquivo no início do método, depois faz todo o processamento das tags e só no final que o CSV é usado de fato.

Se esta função crescer, com o código agrupado fica mais fácil de entender no futuro.

Reply:
Deu pra perceber @VictorAssis!
Claro isso deixa o codigo bem manutencionavel, sendo que os emails lidos so estao sendo utilizados no finalzinho mesmo!

Reply:
Deixei nessa posição por causa do código anterior, vou mandar um commit resolvendo isso.